File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys
input = sys.stdin.readline
n = int(input())
meeting = []
result = 0
ending = 0

for i in range(n):
    a, b = map(int, input().split())
    meeting.append((a, b))
meeting = sorted(meeting, key=lambda x: (x[1], x[0]))
for _ in meeting:
    if _[0] >= ending:
        result += 1
        ending = _[1]
print(result)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
//문제: Doit035
//0. 문제 이해 :
//1. 시간 복잡도 확인:
//2. 적용 알고리즘 확인:
//3. 맞는 자료구조 확인:
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");

solution(input);
function solution(input) {
    const N = input.shift();
    const ARR = input.map((e) => e.split(" ").map(Number));

    ARR.sort((a, b) => {
        if (a[1] === b[1]) {
            return a[0] - b[0];
        } else return a[1] - b[1];
    });
    let end = 0;
    let answer = 0;
    for (let meet of ARR) {
        if (meet[0] >= end) {
            answer++;
            end = meet[1];
        }
    }
    console.log(answer);
    // console.log(ARR);
}